src/main/webapp/gast/stat.jsp

```diff
@@ -0,0 +1,155 @@
+    }
+
+    public String html_sort_title_down(int page, String filterTitle, int recordsPerPage, String jumpToID) {
+        return "<a class=\"sort-link\"href=\"?jumpToID="+jumpToID+"&sort=titleDown&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + page + "\">sort down</a>";
+    }
+
+    public void print_pagination(JspWriter out, int currentPage, int recordsPerPage, String filterTitle, int nOfPages, String sort, String jumpToID) throws Exception {
+        out.println("<nav aria-label=\"Navigation for rows\">");
+        out.println("<ul class=\"pagination\">");
+
+        /*Print Previous Button*/
+        if (currentPage != 1) {
+            out.println(html_prev_button(filterTitle, String.valueOf(recordsPerPage), currentPage,sort,jumpToID));
+        }
+
+        /*Print Pages and highlight current page*/
+        for (int i = 1; i <= nOfPages; i++) {
+            if (currentPage == i) {
+                out.println(html_page_item_current(i));
+            } else {
+                out.println(html_page_item(i, filterTitle, recordsPerPage,sort,jumpToID));
+            }
+        }
+
+        /*Print Next Button*/
+        if (currentPage < nOfPages) {
+            out.println(html_next_button(filterTitle, String.valueOf(recordsPerPage), currentPage,sort,jumpToID));
+        }
+        out.println("</ul>");
+        out.println("</nav>");
+    }
+%>
+<p>
+        
+    <h1>Statistik</h1>
+    <a href="/neg/gast/stat">Zurück zur Übersicht</a>
+    <br><br>
+    <h3>Liste der Quellen mit Anzahl der Belege
+    </h3>
+    <br>
+    <%
+        print_pagination(out, currentPage, recordsPerPage, filterTitle, nOfPages,sort,jumpToID);
+    %> 
+    <table id="stat1" class="statTable">
+        <thead>        
+            <th>
+                <b>Titel der Quelle</b>
+                <form method="GET">
+                    <input name="filterTitle" type="text" size="40" value="<%=filterTitle %>" placeholder="Titel Filter"/>
+                    <input name="page" type="hidden" value="anzahl_belege"/>
+                    <input name="sort" type="hidden" value="<%=sort %>"/>
+                    <input type="submit" />
+                </form>
+                <%
+                    out.println(html_sort_title_up(currentPage, filterTitle, recordsPerPage,jumpToID));
+                    out.println(html_sort_title_down(currentPage, filterTitle, recordsPerPage,jumpToID));
+                %>
+            </th>
+            <th>
+                <b>Anzahl Belege</b>
+            </th>
+        </thead>
+       <tbody>
+            <%
+            for (Quelle q : lst) {
+                out.print("<tr onclick=\"window.location='/neg/gast/quelle?ID="+String.valueOf(q.getId())+"';\" style=\"cursor: pointer;\">");
```

Comment:
The full path including /neg should never be hardcoded because this wont work on the servers. Please build the URL using the Utils.getBaseUrl() prefix instead.

Reply:
Also, instead of the onclick event please use a proper hyperlink <a>...</a> instead inside the first <td> of your table. This will not only mark it in a way that is visible to the user, but also prevent CSP (Content Security Policy) issues. onclick should never be used in any context.

Reply:
fixed!